### ISSUE TITLE
Prepared select statement crashes with System.OverflowException

### ISSUE THREAD

**Issue by helmutschneider** [mysql-net/MySqlConnector#1064]
MySqlConnector crashes sporadically on various simple select statements. I assume the driver encounters some unexpected binary sequence in the result set. It only happens when the statement is prepared.

.NET runtime: 5.0.401
Connector version: 1.3.13
MySQL version: 8.0.18, 8.0.27
OS: MacOS 10.15, Google Cloud SQL

Repro:
```csharp
using System;
using System.Data.Common;
using System.Threading.Tasks;
using MySqlConnector;

namespace db_bug
{
    class Program
    {
        static readonly MySqlConnectionStringBuilder Local = new()
        {
            Server = "127.0.0.1",
            Database = "720",
            UserID = "root",
            Password = "",
            CharacterSet = "utf8mb4",
            SslMode = MySqlSslMode.None,
            DateTimeKind = MySqlDateTimeKind.Utc,
        };

        static async Task Main(string[] args)
        {
            using var conn = new MySqlConnection(Local.ConnectionString);
            await conn.OpenAsync();

            // you can fiddle with these numbers to make the crash go away.
            var query = @"
            SELECT 23806 AS a,
                   1 AS b,
                   238729 AS c
            ";

            await SelectAsync(conn, query);

            Console.WriteLine("OK");
        }

        static async Task SelectAsync(DbConnection conn, string query)
        {
            using var stmt = conn.CreateCommand();
            stmt.CommandText = query;

            // only crashes if we prepare the statement.
            await stmt.PrepareAsync();

            using var rdr = await stmt.ExecuteReaderAsync();
            while (await rdr.ReadAsync())
            {
                // discard!
            }
        }
    }
}

```

```
Unhandled exception. System.OverflowException: Arithmetic operation resulted in an overflow.
   at MySqlConnector.Core.ResultSet.<ScanRowAsync>g__ScanRowAsyncRemainder|9_1(ResultSet resultSet, PayloadData payload, Row row) in /Users/johan/Desktop/MySqlConnector/src/MySqlConnector/Core/ResultSet.cs:line 316
   at MySqlConnector.Core.ResultSet.ScanRowAsync(IOBehavior ioBehavior, Row row, CancellationToken cancellationToken) in /Users/johan/Desktop/MySqlConnector/src/MySqlConnector/Core/ResultSet.cs:line 236
   at MySqlConnector.Core.ResultSet.ReadAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in /Users/johan/Desktop/MySqlConnector/src/MySqlConnector/Core/ResultSet.cs:line 200
   at MySqlConnector.MySqlDataReader.ReadAsync(CancellationToken cancellationToken) in /Users/johan/Desktop/MySqlConnector/src/MySqlConnector/MySqlDataReader.cs:line 42
   at db_bug.Program.SelectAsync(DbConnection conn, String query) in /Users/johan/Desktop/db_bug/Program.cs:line 46
   at db_bug.Program.Main(String[] args) in /Users/johan/Desktop/db_bug/Program.cs:line 32
   at db_bug.Program.<Main>(String[] args)
```

**Comment by bgrainger** [comment]
See https://github.com/mysql-net/MySqlConnector/issues/1018. There is not currently a fix for this in a 1.3.x version. You could try upgrading to 2.0.0-beta.5, or (as a workaround) add `IgnorePrepare=true;` to your connection string to disable `MySqlCommand.Prepare`.

**Comment by bgrainger** [comment]
Duplicate of #1018
